src/amo/components/AutoSearchInput/index.js

```diff
@@ -162,10 +163,20 @@ export class AutoSearchInputBase extends React.Component<Props, State> {
       return;
     }
 
+    if (value.length < SEARCH_TERM_MIN_LENGTH) {
+      log.debug(oneLine`Ignoring suggestions fetch because query
+      subceeds min length (${SEARCH_TERM_MIN_LENGTH})`);
+
+      this.props.dispatch(autocompleteCancel());
+      return;
+    }
+
     if (value.length > SEARCH_TERM_MAX_LENGTH) {
       log.debug(oneLine`Ignoring suggestions fetch because query
         exceeds max length (${SEARCH_TERM_MAX_LENGTH})`
       );
+
+      this.props.dispatch(autocompleteCancel());
```

Comment:
The [test for long queries](https://github.com/SeanPrashad/addons-frontend/blob/885452970d79f31ab8793bd6b62f51e21dbcbe4c/tests/unit/amo/components/TestAutoSearchInput.js#L322) also needs to make sure that the cancel action was dispatched.